src/Juvix/Pipeline/Internal.hs

```diff
@@ -0,0 +1,61 @@
+module Juvix.Pipeline.Internal where
+
+import qualified Data.HashMap.Strict as HM
+import qualified Juvix.Core as Core
+import qualified Juvix.Core.Common.Context as Context
+import qualified Juvix.Core.Common.Context.Traverse as Context
+import qualified Juvix.Core.IR.Types as IR
+import qualified Juvix.Core.Parameterisation as P
+import qualified Juvix.Frontend as Frontend
+import Juvix.Library
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import Juvix.Library.Parser (ParserError)
+import qualified Juvix.Library.Sexp as Sexp
+import qualified Juvix.ToCore.FromFrontend as FF
+import Prelude (String)
+
+data Error
+  = PipeLine Core.Error
+  | ParseErr ParserError
+  deriving (Show)
+
+toCore :: [FilePath] -> IO (Either Error (Context.T Sexp.T Sexp.T Sexp.T))
```

Comment:
I think we want these functions to use `Feedback`/`Pipeline` as well, but that should be done in a separate issue as well.